hal/network/ncp/wifi/platform_ncp_update.cpp

```diff
@@ -114,14 +163,10 @@ int platform_ncp_fetch_module_info(hal_system_info_t* sys_info, bool create) {
         hal_module_t* module = sys_info->modules + i;
         if (!memcmp(&module->bounds, &module_ncp_mono, sizeof(module_ncp_mono))) {
             if (create) {
-                const auto ncpClient = particle::wifiNetworkManager()->ncpClient();
-                SPARK_ASSERT(ncpClient);
-                CHECK(ncpClient->on());
-                uint16_t version;
-                int error = ncpClient->getFirmwareModuleVersion(&version);
-                if (error) {
-                    version = 0;
-                }
+                uint16_t version = 0;
+                // Defaults to zero in case of failure
+                getWifiNcpFirmwareVersion(&version);
```

Comment:
How about do this in the `Esp32NcpNetif::loop` before entering the while loop like we do to the Argon to fetch the MAC address? Just to be consistence to initialize things in one place.

Reply:
This method will get called multiple times throughout the device runtime, so this is the appropriate place to look into cache first and then fallback to calling into ESP32 itself to get the version (and update the cache).

If we are thinking of checking/updating the cache on early boot as we do with a MAC address on an Argon, this is disabled on Tracker already, so that we don't unnecessarily boot ESP32 every time we boot Device OS. Eventually using the caching that was implemented here we'll do the same for MAC address. So, I'd prefer to keep things on as-needed basis.